### PR TITLE
NGC Filesystem Improvements

### PR DESCRIPTION
**Pull request by NickGeneva**
# Modulus Pull Request

Some improvements to the NGC model function:
- Adding support for organization level models
- Adding informative error for next gen of API keys
- Make unzip functionality only apply for .zip files (fixing unzipping of mdlus files)

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: modulus/utils/filesystem.py [hunk 1]
```diff
@@ -69,26 +71,38 @@ def _download_ngc_model_file(path: str, out_path: str, timeout: int = 300) -> st
...
-    (org, team, model_version, filename) = path.split("/", 4)
-    (model, version) = model_version.split("@", 1)
+    if len(path.split("@")[0].split("/")) == 3:
+        (org, team, model_version, filename) = path.split("/", 3)
+        (model, version) = model_version.split("@", 1)
+    else:
+        (org, model_version, filename) = path.split("/", 2)
+        (model, version) = model_version.split("@", 1)
+        team = None
+
```
Comment (dallasfoster) [inline modulus/utils/filesystem.py:88]: Why is it that we make the users submit the path with the model, team (optional), filename, model built in (which they would have to do manually anyways) just for us to then split it apart anyways? Would it not make sense to have the api look something like this:

```python
_download_ngc_model_File(
    org, 
    model_id, 
    version, 
    file_path, 
    team = None, ...
    ):
```

**Review by dallasfoster — APPROVED** [review]

File: modulus/utils/filesystem.py (hunk 1 above)
Comment (NickGeneva) [reply modulus/utils/filesystem.py:88]: This is integrated into the file system, so it makes sense to use a path. Follows hugging face for url structure.

**Review by loliverhennigh — APPROVED** [review]
LGTM

**Pull request merged**